src/tokenize.c

```diff
@@ -72,6 +69,73 @@ static char *DefaultNormalize(char *s, char *dst, size_t *len) {
   return dst;
 }
 
+static char *DefaultNormalize_utf8(char *s, char *dst, size_t *len) {
+  // TODO: set locale depending on the index language?
+  setlocale(LC_ALL, "en_US.UTF-8");
+
+  size_t origLen = *len;
+  char *realDest = s;
+  size_t dstLen = 0;
+
+#define SWITCH_DEST()        \
+  if (realDest != dst) {     \
+    realDest = dst;          \
+    memcpy(realDest, s, ii); \
+  }
+  // set to 1 if the previous character was a backslash escape
+  int escaped = 0;
+  mbstate_t state;
+  memset(&state, 0, sizeof(state));
+  wchar_t wc;
+  size_t len_wc;
+
+  for (size_t ii = 0; ii < origLen;) {
+    len_wc = mbrtowc(&wc, &s[ii], MB_CUR_MAX, &state);
+    if (len_wc == (size_t)-1 || len_wc == (size_t)-2) {
+      // Handle invalid multi-byte sequence
```

Comment:
Is there a simple example for this that we can test (for coverage), or is this a highly edge-case?

Reply:
I added tests with invalid sequences, but those cases are not seen there. What if we add an assertion?

Reply:
Do you have any idea when this can happen?